src/DataStructures/Variables.hpp

```diff
@@ -813,6 +814,26 @@ void swap(Variables<tmpl::list<TagsLhs...>>& lhs,
   rhs = std::move(temp);
 }
 
+namespace detail {
+
+template <typename F, typename TagsList, typename... ApplyTags>
+constexpr decltype(auto) apply_impl(F&& f, const Variables<TagsList>& t,
+                                    tmpl::list<ApplyTags...> /* meta */) {
+  return std::forward<F>(f)(get<ApplyTags>(t)...);
+}
+
+}  // namespace detail
+
+template <typename ApplyTags, typename F, typename TagsList>
+constexpr decltype(auto) apply(F&& f, const Variables<TagsList>& t) {
+  return detail::apply_impl(std::forward<F>(f), t, ApplyTags{});
+}
+
+template <typename F, typename TagsList>
+constexpr decltype(auto) apply(F&& f, const Variables<TagsList>& t) {
```

Comment:
Oh I didn't mean to push this hunk, it's from some earlier experimentation. Dropped it now. However, I think it could be useful to make the interface our tagged containers more consistent (`tags_list` alias, `apply` function etc.)